### ISSUE TITLE
Accessibility: Duplicative label Announcement

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#55416]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The fix to https://github.com/woocommerce/woocommerce/issues/51794 in https://github.com/woocommerce/woocommerce/pull/52949 added a duplicative label announcement for screen reader users.

### Recommended Fix
Remove the "ID" referring to the `<label>` from the aria-describedby attributes on each radio input. The only ID that should be referenced by the aria-describedby attribute is the ID for the description. The label is already associated with the input because the `<label>` tag is wrapped around the input.

### Expected behavior

The label will only be read once and then the description will be read out.

### Actual behavior

The label is read twice by NVDA because it is read as the label and is also referenced in the aria-describedby attribute on the input. 

For instance, the Cash on Delivery input reads as "Cash on delivery radio button checked Cash on
delivery Pay with cash upon delivery. 3 of 3".

#### Screenshot 
<img width="2174" alt="Image showing NVDA output on Cash on Deliver filed" src="https://github.com/user-attachments/assets/ac9f5c28-723c-4297-88fe-ee3a19cadb1f" />

#### Code causing the issue:
```
<div class="wc-block-components-radio-control-accordion-option wc-block-components-radio-control-accordion-option--checked-option-highlighted">
   <label class="wc-block-components-radio-control__option wc-block-components-radio-control__option-checked" for="radio-control-wc-payment-method-options-cod">
      <input id="radio-control-wc-payment-method-options-cod" class="wc-block-components-radio-control__input" type="radio" name="radio-control-wc-payment-method-options" aria-describedby="radio-control-wc-payment-method-options-cod__label radio-control-wc-payment-method-options-cod__content" aria-disabled="false" value="cod">
      <div class="wc-block-components-radio-control__option-layout">
         <div class="wc-block-components-radio-control__label-group">
            <span id="radio-control-wc-payment-method-options-cod__label" class="wc-block-components-radio-control__label">
               <span class="wc-block-components-payment-method-label">Cash on delivery</span>
            </span>
         </div>
      </div>
   </label>
   <div id="radio-control-wc-payment-method-options-cod__content" class="wc-block-components-radio-control-accordion-content"
      <div>Pay with cash upon delivery.</div>
   </div>
</div>
```

### Steps to reproduce

1. Add a product the cart.
2. Go to the checkout page.
3. Turn on the NVDA or VoiceOver screen reader.
4. Listen to the radio inputs in the payment method section.

Alternatively, you can also review the code for the aria-describedby attribute on the input here: https://github.com/woocommerce/woocommerce/blob/71f0b7201d0df436affc785d79c5bdbbb854818e/plugins/woocommerce-blocks/packages/components/radio-control/option.tsx#L49

### WordPress Environment

N/A

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by Manussakis** [comment]
Thanks @amberhinds for creating this issue.

For clarification, according to my investigation, the https://github.com/woocommerce/woocommerce/pull/52949 PR didn't actually introduce the issue but was incomplete, since the fix in the new https://github.com/woocommerce/woocommerce/pull/55503 PR could have been done there.
